tests/torchtune/modules/peft/test_lora.py

```diff
@@ -97,6 +110,12 @@ def test_forward(self, inputs, lora_linear, out_dim) -> None:
         assert actual.shape == (BSZ, SEQ_LEN, out_dim)
         torch.testing.assert_close(actual.mean(), expected, atol=1e-4, rtol=1e-6)
 
+    def test_dora_forward(self, inputs, dora_linear, out_dim) -> None:
+        expected = torch.tensor(EXPECTED_VAL)
```

Comment:
Does this mean the expected val of DoRA is the same as LoRA? Why is this, and intuitively, I'm not sure if I understand if the result is the exact same, how DoRA results in different training than LoRA? Pretty sure I'm missing something basic here but would be good to clarify.

Reply:
Here is the explanation:
$$DORA = W*x + (\frac{m}{weight norm} -1)*W *x  + \frac{m}{weight norm} * {lora}_b({lora}_a(x)) + {scaling}$$

The m vector is initialized as self.m == weight norm. So the ratio $\frac{m}{weight norm}$ is 1 for the first iteration.
so LORA == DORA for the first pass.

Better explanation from the author: https://github.com/huggingface/peft/pull/1474#issuecomment-1964581081